src/levanter/models/attention.py

```diff
@@ -57,27 +59,77 @@ def dot_product_attention(
         raise ValueError("QPos and KPos must be different")
 
     if use_flash:
-        from levanter.models.flash_attention import BLOCK_SIZE, flash_attention
-
-        if flash_block_size is None:
-            flash_block_size = BLOCK_SIZE
-
-        return flash_attention(
-            QPos,
-            KPos,
-            Key,
-            query,
-            key,
-            value,
-            block_size=flash_block_size,
-            mask=mask,
-            bias=bias,
-            dropout=dropout,
-            inference=inference,
-            key=prng,
-            dtype=attention_dtype,
-            precision=precision,
-        )
+        accelerator_type = xla_bridge.get_backend().platform.lower()
+
+        if accelerator_type == "tpu" or accelerator_type == "cpu":
+            from levanter.models.flash_attention import flash_attention
+
+            if flash_block_size is None:
+                seq_len = query.shape["position"]
+                flash_block_size = seq_len
+
+            return flash_attention(
+                QPos,
+                KPos,
+                Key,
+                query,
+                key,
+                value,
+                block_size=flash_block_size,
+                mask=mask,
+                bias=bias,
+                dropout=dropout,
+                inference=inference,
+                key=prng,
+                dtype=attention_dtype,
```

Comment:
leave a TODO about what to do for dtype and precision in fused_attn. (For precision I think it's just warn if not default, but I'm not sure)